diff -up WindowMaker-0.95.8/src/window.c.3~ WindowMaker-0.95.8/src/window.c
--- WindowMaker-0.95.8/src/window.c.3~	2019-01-19 22:59:30.363129092 +0100
+++ WindowMaker-0.95.8/src/window.c	2019-01-19 23:01:08.366467958 +0100
@@ -1165,9 +1165,6 @@ WWindow *wManageWindow(WScreen *scr, Win
 	 * related variables (frame_x,frame_y) */
 	wWindowConfigure(wwin, x, y, width, height);
 
-	/* to make sure the window receives it's new position after reparenting */
-	wWindowSynthConfigureNotify(wwin);
-
 	/* Setup descriptors and save window to internal lists */
 	if (wwin->main_window != None) {
 		WApplication *app;
@@ -1283,6 +1280,9 @@ WWindow *wManageWindow(WScreen *scr, Win
 	if (title)
 		XFree(title);
 
+	/* to make sure the window receives it's new position after reparenting */
+	wWindowSynthConfigureNotify(wwin);
+
 	XUngrabServer(dpy);
 
 	/* Final preparations before window is ready to go */
