diff -up WindowMaker-0.95.8/WindowMaker/Defaults/WMState.in.4~ WindowMaker-0.95.8/WindowMaker/Defaults/WMState.in
--- WindowMaker-0.95.8/WindowMaker/Defaults/WMState.in.4~	2019-01-22 03:14:38.272558141 +0100
+++ WindowMaker-0.95.8/WindowMaker/Defaults/WMState.in	2019-01-22 03:15:43.145477069 +0100
@@ -10,11 +10,21 @@
         Lock = Yes;
       },
       {
-	Command = xterm;
+	Command = xvt;
 	Name = xterm.XTerm;
 	AutoLaunch = No;
 	Forced = No;
 	Position = "0,1";
+      },
+      {
+        Command = "wmcalclock -b 100 -24 -tc #001100 -bc #7e9e69";
+        Name = "wmcalclock.wmcalclock";
+        AutoLaunch = Yes;
+        Lock = No;
+        Forced = No;
+        BuggyApplication = No;
+        Position = "0,3";
+        DropCommand = "wmcalclock -b 100 -24 -tc #001100 -bc #7e9e69 %d";
       }
     );
     Position = "-64,0";
