diff -up WindowMaker-0.95.8/src/wmspec.c.5~ WindowMaker-0.95.8/src/wmspec.c
--- WindowMaker-0.95.8/src/wmspec.c.5~	2019-01-22 03:16:01.845742359 +0100
+++ WindowMaker-0.95.8/src/wmspec.c	2019-01-22 03:17:15.349786683 +0100
@@ -1652,6 +1652,11 @@ Bool wNETWMProcessClientMessage(XClientM
 			wWindowChangeWorkspace(wwin, desktop);
 		}
 		return True;
+
+	} else if (event->message_type == net_wm_moveresize) {
+		/* TODO, but accept it anyway */
+		
+		XUngrabPointer(dpy, CurrentTime);
 	}
 
 	return False;
